src/modules/user-group/services/user-group.service.ts

```diff
@@ -0,0 +1,230 @@
+		} catch (err) {
+			throw new CustomError('Failed to fetch user groups from graphql', err, {
+				variables,
+				query: 'GET_USER_GROUPS_WITH_FILTERS',
+			});
+		}
+	}
+
+	public static async fetchAllUserGroups(): Promise<UserGroup[]> {
+		const response = await UserGroupService.fetchUserGroups(0, 'label', 'asc', {});
+		return response[0];
+	}
+
+	public static async fetchUserGroupById(id: string): Promise<UserGroup | undefined> {
+		let variables: any;
+		try {
+			variables = {
+				id,
+			};
+			const response = await dataService.query({
+				variables,
+				query: this.getQueries().GetUserGroupByIdDocument,
+			});
+
+			if (response.errors) {
+				throw new CustomError('response contains errors', null, { response });
+			}
+
+			return get(response, 'data.users_groups[0]') || get(response, 'data.users_group[0]');
+		} catch (err) {
+			throw new CustomError('Failed to fetch user group by id from graphql', err, {
+				variables,
+				query: 'GET_USER_GROUP_BY_ID',
+			});
+		}
+	}
+
+	// TODO convert to addPermissionToUserGroup
+	// public static async addPermissionGroupsToUserGroup(
+	// 	permissionGroupIds: number[],
+	// 	userGroupId: number | string
+	// ): Promise<void> {
+	// 	try {
+	// 		const response = await dataService.mutate({
+	// 			mutation: ADD_PERMISSION_GROUPS_TO_USER_GROUP,
+	// 			variables: {
+	// 				objs: permissionGroupIds.map((permissionGroupId) => ({
+	// 					user_permission_group_id: permissionGroupId,
+	// 					user_group_id: userGroupId,
+	// 				})),
+	// 			},
+	// 			update: ApolloCacheManager.clearUserGroupCache,
+	// 		});
+	// 		if (response.errors) {
+	// 			throw new CustomError('Failed to add permission groups to user group', null, {
+	// 				errors: response.errors,
+	// 			});
+	// 		}
+	// 	} catch (err) {
+	// 		throw new CustomError('Failed to add permission groups to user group', err, {
+	// 			query: 'ADD_PERMISSION_GROUPS_TO_USER_GROUP',
+	// 			variables: {
+	// 				permissionGroupIds,
+	// 				userGroupId,
+	// 			},
+	// 		});
+	// 	}
+	// }
+
+	// TODO convert to removePermissionsFromUserGroup
+	// public static async removePermissionGroupsFromUserGroup(
+	// 	permissionGroupIds: number[],
+	// 	userGroupId: number | string
+	// ): Promise<void> {
+	// 	try {
+	// 		const response = await dataService.mutate({
+	// 			mutation: REMOVE_PERMISSION_GROUPS_FROM_USER_GROUP,
+	// 			variables: {
+	// 				permissionGroupIds,
+	// 				userGroupId,
+	// 			},
+	// 			update: ApolloCacheManager.clearUserGroupCache,
+	// 		});
+	// 		if (response.errors) {
+	// 			throw new CustomError('Failed to remove permission groups from user group', null, {
+	// 				errors: response.errors,
+	// 			});
+	// 		}
+	// 	} catch (err) {
+	// 		throw new CustomError('Failed to remove permission groups from user group', err, {
+	// 			query: 'REMOVE_PERMISSION_GROUPS_FROM_USER_GROUP',
+	// 			variables: {
+	// 				permissionGroupIds,
+	// 				userGroupId,
+	// 			},
+	// 		});
+	// 	}
+	// }
```

Comment:
are these still needed, or can they be removed?

Reply:
Removed.